src/components/AugmentedResultList/AugmentedResultList.ts

```diff
@@ -0,0 +1,142 @@
+import { $$, ComponentOptions, IComponentBindings, Initialization, IQueryResult, IResultListOptions, IFieldOption } from 'coveo-search-ui';
+
+/**
+ * Interface for the object data returned from remote action.
+ *
+ * @export
+ * @interface IObjectData
+ */
+export interface IObjectData {
+    /**
+     * Data specific to a result with matching object id.
+     */
+    objectData: {}[];
+    /**
+     * Data to add to every result with matching object id.
+     */
+    commonData: {};
+}
+
+/**
+ * Generic interface for the response returned by the remote action method.
+ *
+ * @export
+ * @interface IPromiseReturnArgs
+ * @template T
```

Comment:
Where's the template? (that's actually smart to use a template btw, you could templatize IObjectData too so it less loose than '{}')